File: ts/algorithm/sort/quick.ts
export function quick(arr: number[]): number[] {
  if (arr.length <= 1) return arr;

  const base = arr[0];

  const left: number[] = [];
  const right: number[] = [];

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < base) {
      left.push(arr[i]);
    } else {
      right.push(arr[i]);
    }
  }

  return [...quick(left), base, ...quick(right)];
}

File: python/handwrite/algorithms/sort/quick.py
def quick(arr: list[int]):
    if len(arr) <= 1:
        return arr
    base = arr[0]

    left = []
    right = []

    for i in range(1, len(arr)):
        if arr[i] < base:
            left.append(arr[i])
        else:
            right.append(arr[i])

    b = quick(left)
    b.append(base)
    b.extend(quick(right))
    return b


print(quick([4, 234, 645, 214, 324, 567, 2, 341, -1]))
